plasmapy/plasma/equilibria1d.py

```diff
@@ -0,0 +1,108 @@
+"""Functionality for representing one-dimensional equilibria."""
+
+__all__ = ["HarrisSheet"]
+
+import astropy.constants as const
+import astropy.units as u
+import numpy as np
+
+from plasmapy.utils.decorators.validators import validate_quantities
+
+
+class HarrisSheet:
+    """
+    Define a Harris Sheet Equilibrium.
+    Magnetic field will be in the :math:`\pm x` direction and
+    the current density will be in the :math:`\pm z` direction,
+    :math:`\hat{x} \times \hat{y} = \hat{z}` coordinate system.
+A Harris sheet is a 1D ideal MHD equilibrium. In resistive MHD if there is any resistivity, it won't be a true equilibrium since the resistivity will gradually smooth the profile out over time. A Harris sheet is often used as the initial condition for simulations of magnetic reconnection.
+
+    Parameters
+    ----------
+    B0 : `~astropy.units.Quantity`
+         Magnitude of magnetic field in the limit of :math:`y → ∞` in units
+         convertible to teslas.
+
+    delta : `~astropy.units.Quantity`
+        The thickness of the current sheet in units convertible to meters.
+
+    P0 : `~astropy.units.Quantity`
+        The plasma pressure in the limit of :math:`y → ∞` in units
+        convertible to pascals.
+
+    Notes
+    -----
+    A current sheet is current limited to a surface.
```

Comment:
```suggestion
    A current sheet is current limited to a surface.

    A Harris sheet is a 1D ideal MHD equilibrium. In resistive MHD if
    there is any resistivity, it won't be a true equilibrium since the
    resistivity will gradually smooth the profile out over time. A 
    Harris sheet is often used as the initial condition for simulations
    of magnetic reconnection.
```
(moved from above)